packages/go_router/doc/redirection.md

```diff
@@ -0,0 +1,30 @@
+Redirection changes the location to a new one based on application state. Redirection can
+be used to display a sign-in screen if the user is not logged in, for example.
+
+A redirect is a callback of the type
+[GoRouterRedirect](go_router/GoRouterRedirect.html). To change incoming location
+based on some application state, add a callback to either the GoRouter or
+GoRoute constructor:
+
+
+```dart
+redirect: (BuildContext context, GoRouterState state) {
+  if (AuthState.of(context).isSignedIn) {
+    return '/signin';
+  } else {
+    return null;
+  }   
+},
+```
```

Comment:
Should we explain what the behavior is if:

* You return `null`
* You return the same location as the current location

Reply:
👍